goss-testing/scripts/iscsi_sbps_sanity.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+#
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# This script is to do some sanity tests for iSCSI based boot content 
+# projection.
+
+# shellcheck disable=SC2207,SC2143,SC2002
+
+Found=0
+LIO_ACT=0
+
+echo "Checking whether the node is configured with iSCSI"
+
+ISCSI=($(kubectl get nodes --selector='iscsi=sbps' -o jsonpath='{.items[*].metadata.name}'))
+
+NODE="$HOSTNAME"
+
+for i in "${ISCSI[@]}"
+do
+    if [ "${i}" == "${NODE}" ]
+    then
+        Found=1
+        echo "$NODE is configured with iSCSI"
+        break
+    fi
+done
+
+if [ "${Found}" == 0 ]
+then
+    echo "iscsi is not configured, exiting"
+    exit 1
+fi
+
+# Verifying whether Marshal agent is running or not
+
+echo "*******************************"
+sbps_marshal=$(systemctl is-active sbps-marshal.service)
+
+if [ "${sbps_marshal}" == 'active' ]
+then
+    echo "Marshal agent is running"
+else
+    echo "Marshal agent is not running"
+fi
```

Comment:
This could be simply:
```
echo "Marshal agent is $(systemctl is-active sbps-marshal.service)"
```
it'll just say `active`/`inactivce` instread of `running`/`not running`.